### PR TITLE
Full node settings -> Functional/tsx, part2

### PR DESCRIPTION
**Pull request by benma**

### CODE REVIEW

File: frontends/web/src/routes/settings/electrum-add-server.tsx [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+        value={electrumCert}
+        placeholder={'-----BEGIN CERTIFICATE-----\n...\n-----END CERTIFICATE-----'}
+      />
+      <div className={[style.block, 'flex flex-row flex-end'].join(' ')}>
+        <Button primary disabled={loadingCert || electrumCert !== ''} onClick={downloadCert}>
+          {
+            loadingCert && (
+              <div className={style.miniSpinnerContainer}>
+                <div className={style.miniSpinner}></div>
+              </div>
```
Comment (shonsirsha) [inline frontends/web/src/routes/settings/electrum-add-server.tsx:112]: As this exact same "loading" component is being used in multiple places (twice in this file, and once in `electrum-server`), it makes sense to extract it into a different file / component.

**Review by shonsirsha — APPROVED** [review]
Tested and worked normally. Just some super minor things, other than that code looks good! 

LGTM after the changes :) 🚀

File: frontends/web/src/routes/settings/server.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export type TElectrumServer = {
+  server: string;
+  tls: boolean;
+  pemCert: string;
+};
```
[the remark this answers is not in the document]
Comment (benma) [reply frontends/web/src/routes/settings/server.ts]: Done

File: frontends/web/src/routes/settings/electrum-server.tsx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+  return (
+    <li>
+      <div className={style.server}>
+        <div className={style.serverLabel}>
+          {server.server}
+          {' '}
+          <strong>{server.tls ? 'TLS' : 'TCP' }</strong>
+        </div>
+        <div>
+          <button className={style.primary} disabled={server.server === '' || (server.tls && server.pemCert === '') || loadingCheck} onClick={check}>
```
[the remark this answers is not in the document]
Comment (benma) [reply frontends/web/src/routes/settings/electrum-server.tsx]: Good idea, done

File: frontends/web/src/routes/settings/electrum-add-server.tsx (hunk 1 above)
Comment (benma) [reply frontends/web/src/routes/settings/electrum-add-server.tsx:112]: Good suggestion - will maybe try that in another PR

Comment (benma) [reply frontends/web/src/routes/settings/electrum-add-server.tsx:112]: Hmm it doesn't work that easily, the buttons have different styles.

**Pull request merged**